rust/cache/src/nop.rs

```diff
@@ -0,0 +1,25 @@
+use std::hash::Hash;
+
+use super::{CacheError, Weighted};
+
+/// A zero-configuration cache that doesn't evict.
+pub struct NopCache;
```

Comment:
I think this is unused? If so, should we remove it?

Reply:
It's wired up.  I think being able to stub out the cache is useful, so I'd prefer to leave it.  You have to set cache config to nop to use it, so the risk of inadvertently using it is low.